### PR TITLE
[unifi] Guest wifi vouchers

### PR DESCRIPTION
**Pull request by mherwege**
This PR adds support for creating guest wifi hotspot vouchers. The PR adds channels to the site thing for showing the next available hotspot voucher and triggering generation for new hotspot vouchers. Channel configuration paramaters allow tailoring the specifications of the vouchers generated.

Signed-off-by: Mark Herwege <mark.herwege@telenet.be>

### CODE REVIEW

File: bundles/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiSiteThingHandler.java [hunk 1]
```diff
@@ -67,32 +72,59 @@ protected boolean initialize(final UniFiSiteThingConfig config) {
...
                 break;
             default:
                 // Unsupported channel; nothing to update
                 return UnDefType.NULL;
         }
-        return new DecimalType(count);
+        return state;
+    }
+
+    private static State countClients(final UniFiSite site, final Function<UniFiClient, Boolean> filter) {
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiSiteThingHandler.java:104]:
```suggestion
    private static State countClients(final UniFiSite site, final Predicate<UniFiClient> filter) {
```

File: bundles/org.openhab.binding.unifi/src/main/resources/OH-INF/config/config.xml [hunk 2]
```diff
@@ -67,6 +67,36 @@
 		</parameter>
 	</config-description>
 
+	<config-description uri="channel-type:unifi:guestVouchersGenerate">
+		<parameter name="voucherCount" type="integer">
+			<label>Number</label>
+			<description>Number of vouchers to create</description>
+			<default>1</default>
+		</parameter>
+		<parameter name="voucherExpiration" type="integer">
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.unifi/src/main/resources/OH-INF/config/config.xml:76]:
```suggestion
		<parameter name="voucherExpiration" type="integer" unit="min">
```

File: bundles/org.openhab.binding.unifi/README.md [hunk 3]
```diff
@@ -113,12 +113,24 @@ The following table describes the `poePort` configuration parameters:
...
+| wirelessClients       | Number    | Number of wireless clients connected                                   | Read        |
+| wiredClients          | Number    | Number of wired clients connected                                      | Read        |
+| guestClients          | Number    | Number of guest clients connected                                      | Read        |
+| guestVoucher          | String    | Guest voucher for access through the guest portal                      | Read        |
+| guestVouchersGenerate | String    | Generate additional guest vouchers for access through the guest portal | Write       |
+
+The `guestVouchersGenerate` string channel is a command only channel that will trigger voucher creation.
+It has configuration parameters to tailor the vouchers created:
+
+* `voucherCount`: Number of vouchers to create (default 1)
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.unifi/README.md:128]: Maybe put the parameters in a table to be more consistent with the other tables, like for `poePort`.

**Review by Hilbrand — COMMENTED** [review]
Just some style comments.

You could also consider adding an action with the parameters. That would make it possible to more dynamically create the vouchers from rules, and/or create virtual items where the user can set the configuration options that are passed when the rule is triggered.

**Comment by mherwege** [comment]
@Hilbrand thank you for the quick review. I incorporated your feedback, but did not implement an action yet. I have not done that before, so would need a bit more time for it. It makes sense though, but probably as an alternative to and on top of the simple solution with already in this PR.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Review by jlaur — APPROVED** [review]
Thanks for adding this feature, looks good to me!

**Pull request merged**